test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/PrimitiveTypesTests/Tests/PrimitiveKeysValuesTests.cs

```diff
@@ -0,0 +1,270 @@
+        public void Int16Test(EdmInt16 entry)
+        {
+            // Arrange & Act
+            var query = _context.CreateQuery<EdmInt16>("EdmInt16Set").Where(e => e.Id.Equals(entry.Id));
+            var queryResult = query.ToArray();
+
+            // Assert
+            //Expected a single result for key value {0}, entry.Id.ToString(CultureInfo.InvariantCulture)
+            Assert.Single(queryResult);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetEdmInt32Set))]
+        public void Int32Test(EdmInt32 entry)
+        {
+            // Arrange & Act
+            var query = _context.CreateQuery<EdmInt32>("EdmInt32Set").Where(e => e.Id.Equals(entry.Id));
+            var queryResult = query.ToArray();
+
+            // Assert
+            //Expected a single result for key value {0}, entry.Id.ToString(CultureInfo.InvariantCulture)
+            Assert.Single(queryResult);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetEdmInt64Set))]
+        public void Int64Test(EdmInt64 entry)
+        {
+            // Arrange & Act
+            var query = _context.CreateQuery<EdmInt64>("EdmInt64Set").Where(e => e.Id.Equals(entry.Id));
+            var queryResult = query.ToArray();
+
+            // Assert
+            //Expected a single result for key value {0}", entry.Id.ToString(CultureInfo.InvariantCulture)
+            Assert.Single(queryResult);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetEdmSingleSet))]
+        public void SingleTest(EdmSingle entry)
+        {
+            // Arrange & Act
+            var query = _context.CreateQuery<EdmSingle>("EdmSingleSet").Where(e => e.Id.Equals(entry.Id));
+            var queryResult = query.ToArray();
+
+            // Assert
+            //Expected a single result for key value {0}, entry.Id.ToString(CultureInfo.InvariantCulture)
+            Assert.Single(queryResult);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetEdmStringSet))]
+        public void StringTest(EdmString entry)
+        {
+            // Arrange & Act
+            var query = _context.CreateQuery<EdmString>("EdmStringSet").Where(e => e.Id.Equals(entry.Id));
+            var queryResult = query.ToArray();
+
+            // Assert
+            //Expected a single result for key value {0} entry.Id)
+            Assert.Single(queryResult);
+        }
+
+        [Theory]
+        [MemberData(nameof(GetEdmTimeSet))]
+        public void TimeTest(EdmTime entry)
+        {
+            // Arrange & Act
+            var query = _context.CreateQuery<EdmTime>("EdmTimeSet").Where(e => e.Id.Equals(entry.Id));
+            var queryResult = query.ToArray();
+
+            // Assert
+            //Expected a single result for key value {0}, entry.Id.ToString()
+            Assert.Single(queryResult);
+        }
+
+        #region TheoryData
+
+        public static TheoryData<T> GetTheoryData<T>(Func<Container, IEnumerable<T>> dataSelector)

```

Comment:
```suggestion
        private static TheoryData<T> GetTheoryData<T>(Func<Container, IEnumerable<T>> dataSelector)

```

Reply:
But do we really need this method. The logic seems complex. Why do we have `IsNotSupportedKey` getting called here? I think we should write the literal keys in `InlineData` to avoid such "magic" in tests.